### PR TITLE
[components] Simplify ScopedField creation

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

As title -- nicer interface that just stacks on top of the existing Pydantic Field class

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26557?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26557** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26557?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26496** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26496?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26495** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26495?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26468** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26468?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by schrockn — COMMENTED** [review]
This is a big bummer as we lose typeahead and typing information. 

Instead I think we should retain the structure with a function that takes a `Field` as an argument, but just with a better name. 

`RenderingScope` or something

**Review by schrockn — CHANGES_REQUESTED** [review]
Per comment

File: python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py [hunk 1]
```diff
@@ -5,20 +5,21 @@
...
 
 REF_BASE = "#/$defs/"
 REF_TEMPLATE = f"{REF_BASE}{{model}}"
 
-CONTEXT_KEY = "required_rendering_context"
+CONTEXT_KEY = "required_rendering_scope"
 
 
-def add_required_rendering_context(field: Any, context: AbstractSet[str]) -> Any:
+def RenderingScope(field: Optional[FieldInfo] = None, *, required_scope: AbstractSet[str]) -> Any:
```
Comment (schrockn) [inline python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py:19]: under what circumstances will `field` be None? Seems simpler/clearer to make it required.

**Review by schrockn — DISMISSED** [review]
back to you for quick question

File: python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py (hunk 1 above)
Comment (OwenKephart) [reply python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py:19]: It's a convenience thing, basically if you have:

```python
class Thing(BaseModel):
    thing: Foo
```
and you want to make it have some required scope, it's nicer to do:

```python
class Thing(BaseModel):
    thing: Foo = RenderingScope(required_scope={"abc"})
```

than

```python
class Thing(BaseModel):
    thing: Foo = RenderingScope(Field(), required_scope={"abc"})
```

Essentially, this lets you ignore the Field() bit in cases where you didn't need to use it before

Comment (schrockn) [reply python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py:19]: roger that makes sense thanks

Would be useful to have these examples in the docblock

**Review by schrockn — APPROVED** [review]

**Comment by OwenKephart** [comment]
### Merge activity

* **Dec 18, 10:52 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/26557).

**Pull request merged**